### ISSUE TITLE
[Discussion]: Re-adding the hold fire command in Delta

### ISSUE THREAD

**Issue by TheGiraffe3** [endless-sky/Endless-Sky-Delta#110]
### Problem Description

As part of updating from ES master, the hold fire command was removed because of some slight bugs.
I personally liked the hold fire command, even though there were some bugs.
This issue should serve as a discussion about whether we want to put it back into ES Delta or not.

### Related Links

https://github.com/endless-sky/endless-sky/pull/9506
https://github.com/endless-sky/endless-sky/issues/10578

### Desired Solution

Re-add the Hold Fire command. It looks like @Hurleveur, who added it originally, deleted the PR branch, so we'd probably have to reimplement the hold fire command by reverting https://github.com/endless-sky/endless-sky/pull/10652.

### Alternative Approaches

N/A

### Additional Context

I don't think the bugs are big enough to be a blocker.

**Comment by Hurleveur** [comment]
It's Amazinite that added it actually
My version had no bugs in it I think (idk what the bug is though tbh), but he took a complicated but cleaner route to it
I can restore the branch easily I think

**Comment by TheGiraffe3** [comment]
The bug was that the gather command no longer worked.
Are you saying Amazinite's version would be better to have in Delta?

**Comment by Zitchas** [comment]
Thanks for doing this. I, too, would like to have a hold-fire command, and if we can do it in a way that doesn't have the bug... That's even better. That being said, if it doesn't have the bug, it might be worth PRing upstream since they presumably wanted it too. The public comments that I've seen are that it was reverted to get rid of the bug, not that they had any wish to get rid of the feature itself.

**Comment by TheGiraffe3** [comment]
That's what it seems.
So my question is whether we're alright with having a slightly buggy version of the Hold Fire command in Delta, or not really, we'll wait until upstream solves it.

**Comment by Hurleveur** [comment]
> The bug was that the gather command no longer worked. Are you saying Amazinite's version would be better to have in Delta?

Hold on I'm looking at it rn
Both version got added that's why I got confused lol, https://github.com/endless-sky/endless-sky/pull/10334 idk if this PR isn't what broke hold position
I restored the branch, feel free to test it, time is short on me unfortunately

**Comment by TheGiraffe3** [comment]
So it looks like we've got an option between two slightly buggy possibilities or not having it at all.
- Escorts can only have one order at once
_or_
- Escorts can't gather and hold at once

We can use just Hurleveur's branch for the first option, or revert the PR that took it out.

**Comment by Zitchas** [comment]
It seems odd that escorts can't gather and hold fire at the same time...

**Comment by TheGiraffe3** [comment]
We could test the changes from https://github.com/endless-sky/endless-sky/pull/10585 here, but warp-core didn't think that would work, and as the author...

**Comment by xobes** [comment]
My 2 cents.  This has got to be doable. But they ought to be able to walk and chew gum at the same time.  Or stop chewing gum.  Or go collect flotsam and don't chew gum.  Or only shoot if shot at.  Or fly away if shot at.

This relates to my own list (see my fork) of wants including a "battle computer" outfit that allows for much better enemy selection and fighter management (vaporware).

😀

**Comment by Zitchas** [comment]
The question is "better enemy selection according to who?"

But yeah, I agree, that this should be doable.

**Comment by xobes** [comment]
> The question is "better enemy selection according to who?"

Me.

one key to choose the closest hostile is not enough.  Very difficult to find the big guy i care about.  Cycling backwards through the list is difficult too.  This is for another topic entirely -- I have a whole list: **https://github.com/xobes/endless-sky/wiki/Idea-list**

**Comment by Zitchas** [comment]
Yes, that makes sense. The hard part is coding into the game to have a shio automatically select the target you prefer. 

I could see a few variants on this, ditching the ship to:
- target strongest hostile
- target weakest hostile
- target closest hostile
- target farthest hostile
